app/bundles/CoreBundle/IpLookup/IP2LocationBinLookup.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2015 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+*/
+
+namespace Mautic\CoreBundle\IpLookup;
+use IP2Location;
+class IP2LocationBinLookup extends AbstractLocalDataLookup
+{
+    /**
+     * @return string
+     */
+    public function getAttribution()
+    {
+        return 'IP2Location Local Bin File DB9BIN only';
+    }
+
+    /**
+     * @return string
+     */
+	 
+    public function getLocalDataStoreFilepath()
+    {  return $this->getDataDir();
+    }
+
+    /**
+     * @return string
+     */
+    public function getRemoteDateStoreDownloadUrl()
+    {
+	   $usernamePass = explode(':',$this->auth);
+	   $data= array();
+	   $data['login'] = $usernamePass[0];
+	   $data['password'] = $usernamePass[1];
```

Comment:
Can you add a check for a valid array with $usernamePass so that it doesn't throw a PHP notice if the user has not set a valid username:password in Configuration. `Notice: Undefined offset: 1`